File: java/socketd/src/main/java/org/noear/socketd/transport/core/FragmentAggregator.java
package org.noear.socketd.transport.core;

import java.io.IOException;

/**
 * 分片聚合器
 *
 * @author noear
 * @since 2.1
 */
public interface FragmentAggregator {
    /**
     * 获取流Id
     */
    String getSid();

    /**
     * 数据流大小
     */
    int getDataStreamSize();

    /**
     * 数据总长度
     */
    int getDataLength();

    /**
     * 添加分片
     */
    void add(int index, MessageInternal message) throws IOException;

    /**
     * 获取聚合帧
     */
    Frame get() throws IOException;
}

File: go/socketd/transport/core/fragment_aggregator.go
package core

import (
	"socketd/transport/core/message"
)

// FragmentAggregator
/**
 * @Description:分片聚合器
 * @Date 2024-01-10 20:57:43
 */
type FragmentAggregator interface {
	GetSid() string                           // 获取流Id
	GetDataStreamSize() int                   // 数据流大小
	GetDataLength() int                       // 数据总长度
	Add(index int, message *message.Message)  // 添加分片
	Get() (frame *message.Message, err error) // 获取聚合帧
}
